### PR TITLE
Document revert strings config

### LINKED ISSUE #234: Document revert strings config
Document the `--revert-strings` flag and the `revert_strings` config value. See https://github.com/foundry-rs/foundry/pull/1377

### PR DESCRIPTION
**Pull request by Sweng0d**
I added --revert strings option on the compiler-options page.

I also explained the possible values of revert string, which are: default, strip, debug and verboseDebug.

Fix #234

### CODE REVIEW

File: src/reference/forge/forge-build.md [hunk 1]
```diff
@@ -87,6 +87,15 @@ optimizerSteps = 'dhfoDgvulfnTUtnIf'
...
 for more information on available settings (specifically `settings.optimizer.details`).
 
+#### Revert Strings
+
+The revert strings is set to default, which means it does not inject compiler-generated revert strings and keeps user-supplied ones.
+
+There are three other options: 
+- Strip: It removes all revert strings (if possible, i.e. if literals are used) keeping side-effects.
+- Debug:  It injects strings for compiler-generated internal reverts, implemented for ABI encoders V1 and V2 for now.
+- verboseDebug: It even appends further information to user-supplied revert strings (not yet implemented).
```
Comment (0xvv) [inline src/reference/forge/forge-build.md:97]: I would suggest removing the "It" at the beginning of these sentences
```suggestion
- Strip: Removes all revert strings (if possible, i.e. if literals are used) keeping side-effects.
- Debug:  Injects strings for compiler-generated internal reverts, implemented for ABI encoders V1 and V2 for now.
- verboseDebug: Appends further information to user-supplied revert strings (not yet implemented).
```

File: src/reference/forge/compiler-options.md [hunk 2]
```diff
@@ -9,6 +9,9 @@
 `--via-ir`  
 &nbsp;&nbsp;&nbsp;&nbsp;Use the Yul intermediate representation compilation pipeline.
 
+`--revert-strings
+&nbsp;&nbsp;&nbsp;&nbsp;How to treat revert (and require) reason strings.
```
Comment (0xvv) [inline src/reference/forge/compiler-options.md:13]: These parenthesis seem unnecessary 
```suggestion
&nbsp;&nbsp;&nbsp;&nbsp;How to treat revert and require reason strings.
```

**Review by 0xvv — COMMENTED** [review]
thanks for doing this, i have a few suggestions !

**Comment by Sweng0d** [comment]
Agreed

File: src/reference/forge/compiler-options.md [hunk 3]
```diff
@@ -9,6 +9,9 @@
 `--via-ir`  
 &nbsp;&nbsp;&nbsp;&nbsp;Use the Yul intermediate representation compilation pipeline.
 
+`--revert-strings
+&nbsp;&nbsp;&nbsp;&nbsp;How to treat revert and require reason strings.
```
Comment (onbjerg) [inline src/reference/forge/compiler-options.md:13]:
```suggestion
`--revert-strings  
&nbsp;&nbsp;&nbsp;&nbsp;How to treat revert and require reason strings.
```

For consistency

File: src/reference/forge/forge-build.md [hunk 4]
```diff
@@ -87,6 +87,15 @@ optimizerSteps = 'dhfoDgvulfnTUtnIf'
 See the [compiler input description documentation](https://docs.soliditylang.org/en/latest/using-the-compiler.html#compiler-input-and-output-json-description)
 for more information on available settings (specifically `settings.optimizer.details`).
 
+#### Revert Strings
+
+The revert strings is set to default, which means it does not inject compiler-generated revert strings and keeps user-supplied ones.
```
Comment (onbjerg) [inline src/reference/forge/forge-build.md:92]:
```suggestion
You can control how revert strings are generated by the compiler. By default, only user supplied revert strings are included in the bytecode, but there are other options:
```

File: src/reference/forge/forge-build.md [hunk 5]
```diff
@@ -87,6 +87,15 @@ optimizerSteps = 'dhfoDgvulfnTUtnIf'
...
+
+There are three other options: 
+- Strip: Removes all revert strings (if possible, i.e. if literals are used) keeping side-effects.
+- Debug:  Injects strings for compiler-generated internal reverts, implemented for ABI encoders V1 and V2 for now.
+- verboseDebug: Appends further information to user-supplied revert strings (not yet implemented).
```
Comment (onbjerg) [inline src/reference/forge/forge-build.md:97]:
```suggestion
- `strip`: Removes all revert strings (if possible, i.e. if literals are used) keeping side-effects.
- `debug`:  Injects strings for compiler-generated internal reverts, implemented for ABI encoders V1 and V2 for now.
- `verboseDebug`: Appends further information to user-supplied revert strings (not yet implemented).
```

**Review by onbjerg — COMMENTED** [review]
minor nits

**Review by onbjerg — APPROVED** [review]

**Comment by onbjerg** [comment]
@all-contributors add @Sweng0d for docs

**Pull request merged**